test/Integration/Porter/PorterTest.php

```diff
@@ -121,11 +121,38 @@ public function testImport()
 
         self::assertInstanceOf(PorterRecords::class, $records);
         self::assertNotSame($this->specification, $records->getSpecification());
+        self::assertInstanceOf(CountableProviderRecords::class, $records->getPreviousCollection());
+        self::assertSame('foo', $records->current());
+    }
+
+    public function testNonCountableIteratorImport()
+    {
+        $porter = (new Porter)->registerProvider(
```

Comment:
There is no need to create a new `Porter` instance for this test. It should be sufficient to simply reprogram the mock provider's `fetch()` method to return a different value as shown in `testImportOneOfNone()` and the tests that follow it.